index.bs

```diff
@@ -217,48 +217,59 @@ Any time an [=XR device=] is needed by an algorithm it can <dfn>ensure an XR dev
+Calling {{XR/supportsSessionMode()}} MUST NOT trigger device-selection UI as this would cause many sites to display XR-specific dialogs early in the document lifecycle without user activation.
 
-The {{XR}} object has an <dfn>active immersive session</dfn>, which MUST be initially <code>null</code>, and a <dfn>list of non-immersive sessions</dfn>, which MUST be initially empty.
+<div class="example">
+The following code checks to see if {{immersive-vr}} sessions are supported.
+
+<pre highlight="js">
+navigator.xr.supportsSessionMode('immersive-vr').then(() => {
+  // 'immersive-vr' sessions are supported.
+  // Page should advertise support to the user.
+}
+</pre>
+</div>
+
+The {{XR}} object has an <dfn>active immersive session</dfn>, which MUST be initially <code>null</code>, and a <dfn>list of inline sessions</dfn>, which MUST be initially empty.
 
 <div class="algorithm" data-algorithm="request-session">
 
 When the <dfn method for="XR">requestSession(|options|)</dfn> method is invoked, the user agent MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
 
   1. [=Ensure an XR device is selected=].
   1. If [=XR device=] is <code>null</code>, [=reject=] |promise| with <code>null</code>.
-  1. If the |options| are not [=supported by the XR device=], [=reject=] |promise| with a {{NotSupportedError}} and abort these steps.
-  1. Let |immersive| be the {{XRSessionCreationOptions/immersive}} attribute of the |options| argument.
-  1. If |immersive| is <code>true</code> and the [=active immersive session=] is not <code>null</code>, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
-  1. If |immersive| is <code>true</code> and the algorithm is not [=triggered by user activation=], [=reject=] |promise| with a {{SecurityError}} and abort these steps.
+  1. Let |mode| be the {{XRSessionCreationOptions/mode}} attribute of the |options| argument.
+  1. If |mode| is in not in [=XR device=]'s [=list of supported modes=], [=reject=] |promise| with a {{NotSupportedError}} and abort these steps.
+  1. If |mode| is {{immersive-vr}} or {{immersive-ar}} and the [=active immersive session=] is not <code>null</code>, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
+  1. If |mode| is {{immersive-vr}} or {{immersive-ar}} and the algorithm is not [=triggered by user activation=], [=reject=] |promise| with a {{SecurityError}} and abort these steps.
   1. Let |session| be a new {{XRSession}}.
   1. [=Initialize the session=] |session| with the [=session description=] given by |options|.
-  1. If |immersive| is <code>true</code> set the [=active immersive session=] to |session|.
-  1. Else append |session| to the [=list of non-immersive sessions=].
+  1. If |mode| is {{immersive-vr}} or {{immersive-ar}} set the [=active immersive session=] to |session|.
```

Comment:
Not specific to this PR, but we need to define the behavior when calling requestSession() twice in rapid succession.  Right now, it looks like the second call wouldn't be rejected unless the first call had already reached this step.

Reply:
Uuuuuuggghhhh. Totally agree that this needs to be addressed (I've been saying I'd do it since May, according to #343), but my goodness the the spec-y algorithm get ugly. 😝 Why can't we just write specs in actual code? 😉